src/tools/cephfs_mirror/Mirror.cc

```diff
@@ -319,19 +319,26 @@ void Mirror::handle_enable_mirroring(const Filesystem &filesystem,
 
   std::scoped_lock locker(m_lock);
   auto &mirror_action = m_mirror_actions.at(filesystem);
-  ceph_assert(mirror_action.action_in_progress);
 
-  mirror_action.action_in_progress = false;
-  m_cond.notify_all();
   if (r < 0) {
     derr << ": failed to initialize FSMirror for filesystem=" << filesystem
          << ": " << cpp_strerror(r) << dendl;
+    // since init failed, don't assert, just unset it directly
+    if (mirror_action.action_in_progress) {
+      mirror_action.action_in_progress = false;
```

Comment:
But this _unset_ isn't directly - its under a condition.

Reply:
i'll remove that condition

Reply:
i'll change the pr name and commit message as well

Reply:
wait, we only unset if its set right, if it's already unset then no need to do anything

Reply:
right -- its not a _direct_ unset is what I meant (as mentioned in the code comment).